pandas/io/json/normalize.py

```diff
@@ -79,62 +85,79 @@ def nested_to_record(ds, prefix="", sep=".", level=0):
             else:
                 newkey = prefix + sep + k
 
+            # flatten if type is dict and
+            # current dict level  < maximum level provided and
             # only dicts gets recurse-flattened
             # only at level>1 do we rename the rest of the keys
-            if not isinstance(v, dict):
+            if (not isinstance(v, dict) or
+                    (max_level is not None and level >= max_level)):
                 if level != 0:  # so we skip copying for top level, common case
                     v = new_d.pop(k)
                     new_d[newkey] = v
                 continue
             else:
                 v = new_d.pop(k)
-                new_d.update(nested_to_record(v, newkey, sep, level + 1))
+                new_d.update(nested_to_record(v, newkey, sep, level + 1,
+                                              max_level))
         new_ds.append(new_d)
 
     if singleton:
         return new_ds[0]
     return new_ds
 
 
-def json_normalize(data, record_path=None, meta=None,
-                   meta_prefix=None,
-                   record_prefix=None,
-                   errors='raise',
-                   sep='.'):
+def json_normalize(data: List[Dict],
+                   record_path: Union[str, List[str]] = None,
+                   meta: Union[str, List, List[str], List[List[str]]] = None,
```

Comment:
Just simplify this to `Union[str, List]`. It makes sense what you are trying to do but this sets an arbitrary depth of 2 dimensions of strings which doesn't really apply.

This simply isn't supported by mypy yet (see https://github.com/python/mypy/issues/731) so OK to just keep as generic List